runtimes/web/src/index.js

```diff
@@ -245,6 +245,62 @@ async function loadCartWasm () {
     window.addEventListener("keydown", onKeyboardEvent);
     window.addEventListener("keyup", onKeyboardEvent);
 
+    let gamepads = 0;
+    function processGamepad() {
+        if (gamepads === 0) {
+            return;
+        }
+
+        const gamepad = navigator.getGamepads()[0];
+        if (!gamepad) {
+
+            return;
+        }
+
+        const buttons = gamepad.buttons;
+        const axes = gamepad.axes;
+
+        // https://www.w3.org/TR/gamepad/#remapping
+        // DPAD + AXIS
+        runtime.maskGamepad(0, constants.BUTTON_UP, buttons[12].pressed || axes[1] < -0.5);
+        runtime.maskGamepad(0, constants.BUTTON_DOWN, buttons[13].pressed || axes[1] > 0.5);
+        runtime.maskGamepad(0, constants.BUTTON_LEFT, buttons[14].pressed || axes[0] < -0.5);
+        runtime.maskGamepad(0, constants.BUTTON_RIGHT, buttons[15].pressed || axes[0] > 0.5);
+
+        // X, O + Triggers
+        // NOTE: for XBox360 a triggers is 6 and 7
+        runtime.maskGamepad(0, constants.BUTTON_X, buttons[0].pressed || buttons[6].pressed);
+        runtime.maskGamepad(0, constants.BUTTON_Z, buttons[1].pressed || buttons[7].pressed);
```

Comment:
Instead of making lots of calls to maskGamepad, let's start with `let gamepad = 0`, assign each bit (eg: `if (buttons[0].pressed) gamepad |= constants.BUTTON_X`) and then apply it with `runtime.setGamepad(0, gamepad)`.

Reply:
This will not working if gamepad buttons will used together with keyboard.
But yep, not make sence, because this is explict case and should be fixed if a multiple gamepads is mapped onto gamepad/keyboard .

Reply:
> This will not working if gamepad buttons will used together with keyboard.

I think that's the case with `maskGamepad` as well since it will clear the button bit if pressed is false.